src/sota_tools/authenticate_test.cc

```diff
@@ -23,36 +26,32 @@ TEST(authenticate, good_zip) {
 }
 
 /* Authenticate with TLS credentials.
- * Parse authentication information from treehub.json.
  * Parse images repository URL from a provided archive. */
-/* TODO: This used to work, but then when the zip file was updated because of
- * expired certs, it was changed to not use cert auth, and there was no check to
- * verify the method at that time.
 TEST(authenticate, good_cert_zip) {
-  boost::filesystem::path filepath = "tests/sota_tools/auth_test_cert_good.zip";
+  // Authenticates with ssl_server on port 1443.
+  boost::filesystem::path filepath = certs_dir / "good.zip";
   ServerCredentials creds(filepath);
-  EXPECT_EQ(creds.GetMethod(), AuthMethod::kCert);
+  EXPECT_EQ(creds.GetMethod(), AuthMethod::kTls);
   TreehubServer treehub;
-  int r = authenticate("", creds, treehub);
+  int r = authenticate("tests/fake_http_server/server.crt", creds, treehub);
```

Comment:
Maybe, makes sense to pass "tests/fake_http_server/server.crt"  as the test parameter ?

Reply:
Sorry, where should it get passed from?

Reply:
From corrsponding CMakeLists.txt

add_aktualizr_test(NAME <test-name> SOURCES <> ARGS <test-args>)

Reply:
but, it's really not important, not sure if it's worth bothering

Reply:
Seems like that's just moving the hardcoded value to a more obscure place. I'm not sure I see the benefit right now. That said, I'm interested in redoing this logic so that we can generate all the certs and such dynamically, and once that is done, something like what you are suggesting will probably be necessary.

Reply:
Yeah, + for the dynamic certs generation, also using TEST_F/test fixtures/base class is an ideal for sharing the same context across multiple tests.